drivers/client_location_history/app/models/client_location_history/location.rb

```diff
@@ -12,16 +12,14 @@ class Location < GrdaWarehouseBase
     belongs_to :place, class_name: 'GrdaWarehouse::Place', primary_key: [:lat, :lon], foreign_key: [:lat, :lon], optional: true
     # this relation isn't used; use polymorphic `source` above
     belongs_to :enrollment, class_name: 'GrdaWarehouse::Hud::Enrollment', optional: true
-    before_create :ensure_grda_warehouse_source
 
     MARKER_COLOR = '#72A0C1'.freeze
 
-    private def ensure_grda_warehouse_source
-      # This somewhat hacky solution gets around the fact that during HMIS Form Processing, we haven't yet saved the
-      # Enrollment being generated, so we don't yet have an ID with which to get the Warehouse enrollment.
-      return unless source_type&.starts_with? 'Hmis::Hud::'
+    # Locations where both `lat` and `lon` are present
+    scope :valid, -> { where.not(lat: nil).and(where.not(lon: nil)) }
```

Comment:
I _think_ the following returns an equivalent query and is a little easier to read:
```suggestion
    scope :valid, -> { where.not(lat: nil, lon: nil) }
```


```
irb(main):002:0> ClientLocationHistory::Location.where.not(lat: nil).and(ClientLocationHistory::Location.where.not(lon: nil)).to_sql
=> "SELECT \"clh_locations\".* FROM \"clh_locations\" WHERE \"clh_locations\".\"lat\" IS NOT NULL AND \"clh_locations\".\"lon\" IS NOT NULL"
irb(main):003:0> ClientLocationHistory::Location.where.not(lat: nil, lon: nil).to_sql
=> "SELECT \"clh_locations\".* FROM \"clh_locations\" WHERE NOT (\"clh_locations\".\"lat\" IS NULL AND \"clh_locations\".\"lon\" IS NULL)"
```

Reply:
We want them both `lat` AND `lon` to be present. I don't think the second query achieves that: `WHERE NOT (\"clh_locations\".\"lat\" IS NULL AND \"clh_locations\".\"lon\" IS NULL)`